js/packages/cli/src/candy-machine-cli.ts

```diff
@@ -100,22 +113,53 @@ programCommand('upload')
       retainAuthority,
       mutable,
       rpcUrl,
+      arweaveJwk,
       batchSize,
     } = cmd.opts();
 
-    if (storage === 'ipfs' && (!ipfsInfuraProjectId || !ipfsInfuraSecret)) {
+    if (storage === StorageType.ArweaveSol && env !== 'mainnet') {
+      throw new Error(
+        'The arweave-sol storage option only works on mainnet. For devnet, please use either arweave, aws or ipfs\n',
+      );
+    }
+
+    if (storage === StorageType.ArweaveBundle && env !== 'mainnet') {
+      throw new Error(
+        'arweave-bundl must receive real AR from their network, for devnet please use AWS or IPFS\n',
```

Comment:
```suggestion
        'The arweave-bundl storage option only works on mainnet because it requires spending real AR tokens. For devnet, please set the --storage option to "aws" or "ipfs"\n',
```